code/player/Inventory.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Linq;
+using Sandbox;
+
+using TTTReborn.Weapons;
+
+namespace TTTReborn.Player
+{
+    partial class Inventory : BaseInventory
+    {
+        public Inventory(TTTPlayer player) : base(player)
+        {
+
+        }
+
+        public override bool Add(Entity entity, bool makeActive = false)
+        {
+            TTTPlayer player = Owner as TTTPlayer;
+
+            if (IsCarryingType(entity.GetType()))
+            {
+                return false;
+            }
+
+            if (entity is Weapon weapon)
+            {
+                Sound.FromWorld("dm.pickup_weapon", entity.Position);
+            }
+
+            bool added = base.Add(entity, makeActive);
+
+            List.Sort((Entity wep1, Entity wep2) => {
+                return (wep1 as Weapon).WeaponType.CompareTo((wep2 as Weapon).WeaponType);
+            });
```

Comment:
`List.Sort((Entity wep1, Entity wep2) => (wep1 as Weapon).WeaponType.CompareTo((wep2 as Weapon).WeaponType));`

Could also remove curly braces, but this works as well.